src/BSON/Binary.stub.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+  * @generate-class-entries static
+  * @generate-function-entries static
+  */
+
+namespace MongoDB\BSON
+{
+    final class Binary implements BinaryInterface, JsonSerializable, Type, \Serializable
```

Comment:
Should `JsonSerializable` be `\JsonSerializable`? It's not defined in the `MongoDB\BSON` namespace.

----

Since `php_phongo_binary_init_ce` includes:

```
#if PHP_VERSION_ID >= 80000
	zend_class_implements(php_phongo_binary_ce, 1, zend_ce_stringable);
#endif
```

Would it make sense to use `#if` here to define two different class declarations, or perhaps just an extra line between this and the opening `{` to avoid duplication? Something like:

```
final class Binary implements BinaryInterface, JsonSerializable, Type, \Serializable
#if PHP_VERSION_ID >= 80000
    , \Stringable
#endif
{
```

Reply:
Yes - I'll change this to `\JsonSerializable`. I was confused by the `php_phongo_json_serializable_ce` variable 🤦 

Unfortunately, the `#if` is completely ignored in the class header (both with and without duplication of the class definition in conditionals). We're unfortunately stuck adding this ourselves until we require PHP 8.0, at which point we can add it to the stub files.

Reply:
See: https://github.com/mongodb/mongo-php-driver/blob/1.14.0/php_phongo.c#L183

We may actually want to continue using `php_phongo_json_serializable_ce`, since it's fetched during MINIT to avoid problems on some distributions where the JSON extension was _not_ compiled in to PHP and instead loaded dynamically. That was definitely an issue back in the 5.x releases (see: [PHPC-878](https://jira.mongodb.org/browse/PHPC-878)), and it may still apply to 7.x+. We may need to confirm with @remicollet.

And while I'm looking at the above, I think the manual fetching of `DateTimeImmutable` may be obsolete. I've opened [PHPC-2115](https://jira.mongodb.org/browse/PHPC-2115) to track that.

Reply:
Json is always build statically in 8.x, so this hack is still useful on 7.x

(btw, only there to avoid "undefined symbol" when load order is not the correct one for linux distribution using `--enable-rtld-now`)